apps/worker/src/app/workflow/usecases/send-message/send-message.base.ts

```diff
@@ -48,7 +56,23 @@ export abstract class SendMessageBase extends SendMessageType {
   protected async getIntegration(
     selectIntegrationCommand: SelectIntegrationCommand
   ): Promise<IntegrationEntity | undefined> {
-    return this.selectIntegration.execute(SelectIntegrationCommand.create(selectIntegrationCommand));
+    const integration = await this.selectIntegration.execute(SelectIntegrationCommand.create(selectIntegrationCommand));
+
+    if (!integration) {
+      return;
+    }
+
+    if (integration.providerId === EmailProviderIdEnum.Novu || integration.providerId === SmsProviderIdEnum.Novu) {
+      integration.credentials = await this.getNovuProviderCredentials.execute({
+        channelType: integration.channel,
+        providerId: integration.providerId,
+        environmentId: integration._environmentId,
+        organizationId: integration._organizationId,
+        userId: selectIntegrationCommand.userId,
+      });
+    }
```

Comment:
so we are doing this because we need the credentials only during the send right? and we don't want to expose it 🙌

Reply:
Yes